core-war/src/main/java/org/silverpeas/web/socialnetwork/myprofil/control/MyProfilSessionController.java

```diff
@@ -242,6 +254,62 @@ public void unlinkSocialNetworkFromSilverpeas(SocialNetworkID networkId) {
     SocialNetworkService.getInstance().removeExternalAccount(getUserId(), networkId);
   }
 
+  public boolean deleteMyAccount(HttpRequest request) {
+    String message = request.getParameter("message");
+    boolean forgetMe = request.getParameterAsBoolean("forgetMe");
+
+    User currentUser = User.getCurrentRequester();
+
+    Administration admin = Administration.get();
+    try {
+      admin.deleteUser(getUserId());
+    } catch (AdminException e) {
+      MessageNotifier.addError(getMultilang().getString("myProfile.actions.deleteAccount.error"));
+      return false;
+    }
+
+    // notify all admins and domain admins
+    if (StringUtil.isDefined(message)) {
+      message = "\"".concat(message).concat("\"");
+    }
+    notifyAdminsToDeletion(message);
+
+    if (forgetMe) {
+      try {
+        admin.blankDeletedUsers(currentUser.getDomainId(), Arrays.asList(getUserId()));
+      } catch (Exception e) {
+        SilverLogger.getLogger(this).error(e);
+      }
+    }
+    return true;
+  }
+
+  private void notifyAdminsToDeletion(final String message) {
+    final OrganizationController organizationController = OrganizationController.get();
+    User currentUser = User.getCurrentRequester();
+    final Domain domain = organizationController.getDomain(currentUser.getDomainId());
+    UserDetailsSearchCriteria criteria = new UserDetailsSearchCriteria()
+        .onUserStatesToExclude(BLOCKED, DEACTIVATED, REMOVED)
+        .onAccessLevels(UserAccessLevel.ADMINISTRATOR);
+    final List<User> admins = organizationController.searchUsers(criteria);
+    criteria = new UserDetailsSearchCriteria()
+        .onDomainIds(currentUser.getDomainId())
+        .onUserStatesToExclude(BLOCKED, DEACTIVATED, REMOVED)
+        .onAccessLevels(UserAccessLevel.DOMAIN_ADMINISTRATOR);
+    admins.addAll(organizationController.searchUsers(criteria));
```

Comment:
These two user search statements for admins can be replaced by just one:
```java
UserDetailsSearchCriteria criteria = new UserDetailsSearchCriteria()
        .onUserStatesToExclude(BLOCKED, DEACTIVATED, REMOVED)
        .onAccessLevels(UserAccessLevel.ADMINISTRATOR, 
                        UserAccessLevel.DOMAIN_ADMINISTRATOR);
final List<User> admins = organizationController.searchUsers(criteria);
```

Reply:
Oups, no in fact. I just see the two statements differ. For one we are looking for platform administrators and the another one we are looking for the domain administrators of the domain to which the user asking for account deletion belongs